gnomad_qc/v3/create_release/prepare_internal_data_release.py

```diff
@@ -0,0 +1,398 @@
+            data_field_name="freq",
+            global_field_name="freq_meta",
+        )
+        freq_ht = freq_ht.transmute(freq=freq_ht.freq.flatmap(lambda x: x.freq))
+        freq_ht = freq_ht.transmute_globals(
+            freq_meta=freq_ht.freq_meta.flatmap(lambda x: x.freq_meta)
+        )
+
+        # Create frequency index dictionary on concatenated array (i.e., including all subsets)
+        # NOTE: non-standard downsampling values are created in the frequency script corresponding to population totals, so new_downsampling values must be evaluated and included in the final freq_index_dict
+        freq_meta = hl.eval(freq_ht.freq_meta)
+        downsamplings = hl.filter(
+            lambda x: x.keys().contains("downsampling"), freq_meta
+        )
+        new_downsamplings = hl.set(
+            hl.map(lambda x: hl.int32(x["downsampling"]), downsamplings)
+        )
+        new_downsamplings_values = hl.eval(
+            new_downsamplings.difference(hl.set(DOWNSAMPLINGS))
+        )
+        DOWNSAMPLINGS.extend([x for x in new_downsamplings_values])
+
+        freq_ht = freq_ht.annotate_globals(
+            freq_index_dict=make_freq_index_dict(freq_ht)
+        )
+
+        if args.test:
+            freq_ht = freq_ht.checkpoint(test_freq_ht_path_concat)
+
+        global_freq_ht = hl.read_table(get_freq().path)
+
+    # Add back in all global frequency annotations not present in concatenated frequencies HT
+    row_fields = set([field for field in global_freq_ht.row_value]).difference(
+        set([field for field in freq_ht.row_value])
+    )
+    logger.info(
+        f"Adding back the following row annotations onto concatenated frequencies: {row_fields}"
+    )
+    freq_ht = freq_ht.annotate(**global_freq_ht[freq_ht.key].select(*row_fields))
+
+    global_fields = set([field for field in global_freq_ht.globals]).difference(
+        set([field for field in freq_ht.globals])
+    )
+    global_fields.remove("downsamplings")
+    logger.info(
+        f"Adding back the following global annotations onto concatenated frequencies: {global_fields}"
+    )
+    freq_ht = freq_ht.annotate_globals(
+        **global_freq_ht.index_globals().select(*global_fields)
+    )
+
+    logger.info("Preparing release Table annotations...")
+    ht = prepare_annotations(freq_ht, args.model_id, test=False)
+
+    logger.info("Removing chrM and sites without filter...")
+    ht = hl.filter_intervals(ht, [hl.parse_locus_interval("chrM")], keep=False)
+    ht = ht.filter(hl.is_defined(ht.filters), keep=True)
+
+    # Add in new annotations for clinical variant interpretation
+    in_silico_ht = hl.read_table(
+        "gs://gnomad/annotations/hail-0.2/ht/genomes_v3.1/gnomad_genomes_v3.1.analyst_annotations.ht"
+    )  # TODO: update path with versioned Table resource path when available
+    ht = ht.annotate(**in_silico_ht[ht.key])
+
+    # Splice in fix to set female metrics to NA on Y chr
+    female_idx = [
```

Comment:
Not sure if I am actually right on this, I just figured it might be best to keep it all in Hail when possible rather than go between python and Hail representation